api/circulation.py

```diff
@@ -585,6 +591,15 @@ def revoke_loan(self, patron, pin, licensepool):
             on_multiple='interchangeable'
         )
         if loan:
+            if not licensepool.open_access:
+                api = self.api_for_license_pool(licensepool)
+                try:
+                    api.checkin(patron, pin, licensepool)
+                except NotCheckedOut, e:
+                    # The book wasn't checked out in the first
+                    # place. Everything's fine.
+                    pass
```

Comment:
I changed this to call the API's `checkin` before deleting the local Loan, because for ODL the Loan's external identifier is needed to return a loan with the API.